### DISCUSSION TITLE
[Interceptor] Support chain of command pattern

### DISCUSSION THREAD

**Discussion by jsaratoga** [dotnet/roslyn#72997] [Ideas]
I currently use metadata and code generators to generate a lot of my application.  The problem comes is when I want to let other others extend code in my system.  I'm current working to implement something sort of like a Chain of Command pattern (like the way d365 works, https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/method-wrapping-coc).  I would really like to see interceptors extended to support
1 - The ability to intercept every call to a specific method without having to specify a file, line number, character
2 - The ability to chain multiple interceptors, the execution order would be random like in D365.

**Comment by CyrusNajmabadi** [comment]
> The ability to intercept every call to a specific method without having to specify a file, line number, character

You could do this by having your own entrypoint call out to an actual impl method.  And then intercept the call *you* make to that impl method to then call some other method.

> The ability to chain multiple interceptors

There is no practical way to do this, so it is nearly certain it will not be done.
